contrib/trace_propagators/src/main/java/io/opentelemetry/contrib/trace/propagation/AwsXRayPropagator.java

```diff
@@ -0,0 +1,266 @@
+
+  private static final Logger logger = Logger.getLogger(AwsXRayPropagator.class.getName());
+
+  private static final char TRACE_HEADER_DELIMITER = ';';
+  private static final char KV_DELIMITER = '=';
+
+  private static final String TRACE_ID_KEY = "Root";
+  private static final int TRACE_ID_LENGTH = 35;
+  private static final String TRACE_ID_VERSION = "1";
+  private static final char TRACE_ID_DELIMITER = '-';
+  private static final int TRACE_ID_DELIMITER_INDEX_1 = 1;
+  private static final int TRACE_ID_DELIMITER_INDEX_2 = 10;
+  private static final int TRACE_ID_FIRST_PART_LENGTH = 8;
+
+  private static final String PARENT_ID_KEY = "Parent";
+  private static final int PARENT_ID_LENGTH = 16;
+
+  private static final String SAMPLED_FLAG_KEY = "Sampled";
+  private static final int SAMPLED_FLAG_LENGTH = 1;
+  private static final char IS_SAMPLED = '1';
+  private static final char NOT_SAMPLED = '0';
+
+  private static final List<String> FIELDS = Collections.singletonList(TRACE_HEADER_KEY);
+
+  @Override
+  public List<String> fields() {
+    return FIELDS;
+  }
+
+  @Override
+  public <C> void inject(Context context, @Nullable C carrier, Setter<C> setter) {
+    Objects.requireNonNull(context, "context");
+    Objects.requireNonNull(setter, "setter");
+
+    Span span = TracingContextUtils.getSpan(context);
+    if (!span.getContext().isValid()) {
+      return;
+    }
+
+    SpanContext spanContext = span.getContext();
+
+    String otTraceId = spanContext.getTraceId().toLowerBase16();
+    String xrayTraceId =
+        TRACE_ID_VERSION
+            + TRACE_ID_DELIMITER
+            + otTraceId.substring(0, TRACE_ID_FIRST_PART_LENGTH)
+            + TRACE_ID_DELIMITER
+            + otTraceId.substring(TRACE_ID_FIRST_PART_LENGTH);
+    String parentId = spanContext.getSpanId().toLowerBase16();
+    char samplingFlag = spanContext.getTraceFlags().isSampled() ? IS_SAMPLED : NOT_SAMPLED;
+    // TODO: Add OT trace state to the X-Ray trace header
+
+    String traceHeader =
+        TRACE_ID_KEY
+            + KV_DELIMITER
+            + xrayTraceId
+            + TRACE_HEADER_DELIMITER
+            + PARENT_ID_KEY
+            + KV_DELIMITER
+            + parentId
+            + TRACE_HEADER_DELIMITER
+            + SAMPLED_FLAG_KEY
+            + KV_DELIMITER
+            + samplingFlag;
+    setter.set(carrier, TRACE_HEADER_KEY, traceHeader);
+  }
+
+  @Override
+  public <C> Context extract(Context context, C carrier, Getter<C> getter) {
+    Objects.requireNonNull(carrier, "carrier");
+    Objects.requireNonNull(getter, "getter");
+
+    SpanContext spanContext = getSpanContextFromHeader(carrier, getter);
+    return TracingContextUtils.withSpan(DefaultSpan.create(spanContext), context);
+  }
+
+  @SuppressWarnings("StringSplitter")
+  private static <C> SpanContext getSpanContextFromHeader(C carrier, Getter<C> getter) {
+    String traceHeader = getter.get(carrier, TRACE_HEADER_KEY);
+    if (Strings.isNullOrEmpty(traceHeader)) {
```

Comment:
```suggestion
    if (traceHeader == null || traceHeader.isEmpty()) {
```

For just this one line I guess we may as well inline this rather than worry about guava or helper methods etc. Alternatively, we don't really need the empty check as an empty string is invalid and will always work with our below logic but it'd be somewhat less optimal. Not sure if a `getter` will return empty strings often enough for us to worry about it though.